pkg/cli/image/extract/extract.go

```diff
@@ -386,28 +386,32 @@ func (o *ExtractOptions) Run() error {
 				}
 
 				srcManifest, location, err := retrieveSourceManifest(ctx, from, repo, o)
-				if err != nil {
-					if err != imagemanifest.AllImageFilteredErr {
-						return err
-					}
+				switch {
+				case err == nil:
 
-					// Translate the current runtime OS to linux when looking through a manifest listed image since a manifest for the particular 'runtime OS'/'archType' may not exist.
-					// In cases where the runtime OS is windows or darwin (i.e. macOS), there is no expectation of a manifest image with those particular OS's.
-					if o.FilterOptions.DefaultOSFilter {
-						runtimeOS := runtime.GOOS
-						arch := runtime.GOARCH
+				// Fail early if err is not recoverable
+				case err != imagemanifest.AllImageFilteredErr:
+					return err
 
-						klog.V(2).Infof("Warning: a manifest image could not be found for this platform: %s/%s. Converting runtime OS, %s, to 'linux' to pull the linux/%s equivalent image.\n", runtimeOS, arch, runtimeOS, arch)
-						o.FilterOptions.OSFilter, err = regexp.Compile("linux/" + arch)
-						if err != nil {
-							return fmt.Errorf("failed to compile OSFilter for linux/%s: %v", arch, err)
-						}
+				// Translate the current runtime OS to linux when looking through a manifest listed image since a manifest for the particular 'runtime OS'/'archType' may not exist.
+				// In cases where the runtime OS is windows or darwin (i.e. macOS), there is no expectation of a manifest image with those particular OS's.
+				case o.FilterOptions.DefaultOSFilter:
+					runtimeOS := runtime.GOOS
+					arch := runtime.GOARCH
 
-						srcManifest, location, err = retrieveSourceManifest(ctx, from, repo, o)
-						if err != nil {
-							return err
-						}
+					klog.V(2).Infof("Warning: a manifest image could not be found for this platform: %s/%s. Converting runtime OS, %s, to 'linux' to pull the linux/%s equivalent image.\n", runtimeOS, arch, runtimeOS, arch)
+					o.FilterOptions.OSFilter, err = regexp.Compile("linux/" + arch)
+					if err != nil {
+						return fmt.Errorf("failed to compile OSFilter for linux/%s: %v", arch, err)
 					}
+
+					srcManifest, location, err = retrieveSourceManifest(ctx, from, repo, o)
```

Comment:
I'm not sure that I successfully chase the fix for the panic (because if-else to switch code changes mask it). Would it make sense return back to if-else instead of switch and only add the the code portion handling panic gracefully?.